golang/client_manager.go

```diff
@@ -90,13 +90,14 @@ type ClientManagerRegistry interface {
 }
 
 type clientManagerRegistry struct {
```

Comment:
Could we remove `clientManagerRegistry`, since each `client` has their own `clientManager`, it seems no need to maintain a "registry".